test/unit/api_spec.js

```diff
@@ -14,22 +14,52 @@
  */
 
 import {
-  createPromiseCapability, FontType, InvalidPDFException, MissingPDFException,
-  PasswordException, PasswordResponses, StreamType
+  createPromiseCapability, FontType, InvalidPDFException, isNodeJS,
+  MissingPDFException, PasswordException, PasswordResponses, StreamType,
+  stringToBytes
 } from '../../src/shared/util';
 import {
   DOMCanvasFactory, RenderingCancelledException
 } from '../../src/display/dom_utils';
-import { PDFDocumentProxy, PDFPageProxy } from '../../src/display/api';
+import {
+  getDocument, PDFDocumentProxy, PDFPageProxy
+} from '../../src/display/api';
+import { NodeFileReader } from './test_utils';
 import { PDFJS } from '../../src/display/global';
 
+const TEST_PDFS_PATH = {
+  dom: '../pdfs/',
+  node: './test/pdfs/',
+};
+
+function buildGetDocumentParams(filename, options) {
+  var params = Object.create(null);
+  if (isNodeJS()) {
+    params.data = new NodeFileReader({
+      path: TEST_PDFS_PATH.node + filename,
+    });
+  } else {
+    params.url = new URL(TEST_PDFS_PATH.dom + filename, window.location).href;
+  }
+  for (var option in options) {
+    params[option] = options[option];
+  }
+  return params;
+}
+
 describe('api', function() {
-  var basicApiUrl = new URL('../pdfs/basicapi.pdf', window.location).href;
+  var basicApiFileName = 'basicapi.pdf';
   var basicApiFileLength = 105779; // bytes
+  var basicApiGetDocumentParams = buildGetDocumentParams(basicApiFileName);
+
   var CanvasFactory;
 
   beforeAll(function(done) {
-    CanvasFactory = new DOMCanvasFactory();
+    if (isNodeJS()) {
+      // TODO: Add `NodeCanvasFactory` to `test_utils.js`.
```

Comment:
Very good catch!
In this case, I forgot to consider that the person writing the comment (i.e. me), has context that a future reader of the code doesn't; I've updated it as below.
```javascript
// NOTE: To support running the canvas-related tests in Node.js,
// a `NodeCanvasFactory` would need to be added (in `test_utils.js`).
```